### ISSUE TITLE
dealer button advancing strategies

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#19]
Currently mb2 assumes that the dealer button is used if and only if there are blinds. We can deal games without blinds (e.g., stud variants), but this code:
```
    fn advance_button(&mut self) {
        if let Some(ref blinds) = self.blind_seats {
```
will not advance the button if there aren't blinds. That's because advancing the button is slightly tricky when you take into consideration people who have bust or have been moved into a table.

The "Limit Hold'em Win the Button" tournament can't be dealt without a different strategy to advance the button.

In addition to the dealer's button, some tournament structures require changing games after some number of orbits, and one way to count orbits is with a button which sometimes can't be the dealer's button because one or more games played might not have one. One of those structures is dealer's choice, which requires a dealer's choice button.

This ticket is to add the code to support "win the button", but  it makes sense for me to think about other buttons while I work on it, in case there's a nice abstraction that covers them all.

**Comment by ctm** [comment]
FWIW, in TESLA:
> During the stud rounds the button will freeze in place.

I need to bear that in mind when I implement this ticket.

**Comment by ctm** [comment]
I've implemented win-the-button and am deploying now. I will take a look at TESLA and verify that "During the stud rounds the button will freeze in place". It's quite possible that doesn't happen, in which case I'll open a bug-report.

I have not done any work associated with tracking orbits for Dealer's Choice. As it is, I only implemented win-the-button because I had a little time left in my work day and figured it would be easy to do (which it was).
